### PR TITLE
Amelia

### PR DESCRIPTION
**Pull request by arcamelia**

### CODE REVIEW

File: src/database/DatabaseConnectionHandler.java [hunk 1]
```diff
@@ -278,4 +298,13 @@ public void update(Model model, int id) {
             }
         }
     }
+
+    public void rollbackConnection() {
```
Comment (anthonylopes54) [inline src/database/DatabaseConnectionHandler.java:302]: Nice

File: src/database/ModelHandler.java [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import java.sql.Connection;
 
 public interface ModelHandler {
-    void Insert(Model model, Connection connection);
+    void insert(Model model, Connection connection);
```
Comment (anthonylopes54) [inline src/database/ModelHandler.java:8]: My bad. Idky I capitalized it

File: src/database/StationHandler.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    // primaryKey: name
+    @Override
+    public void update(Model model, Connection connection) {
+        Station station = (Station) model;
+        String name = station.getName();
+        String address = station.getAddress();
+
+        try {
+            String query = "UPDATE Station SET address = ? WHERE name = ?";
+            PrintablePreparedStatement ps = new PrintablePreparedStatement(connection.prepareStatement(query), query, false);
```
Comment (anthonylopes54) [inline src/database/StationHandler.java:50]: Might need to connect on this. I don't think we need the printablepreparedstatement class

File: src/database/TripHandler.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        } catch (SQLException e) {
+            System.out.println(Constants.EXCEPTION_TAG + " " + e.getMessage());
+            dbHandler.rollbackConnection();
+        }
+    }
+
+    // primaryKey: (seatNum, routeID, passengerID, trainID)
+    @Override
+    public void update(Model model, Connection connection) {
+        Trip trip = (Trip) model;
```
Comment (anthonylopes54) [inline src/database/TripHandler.java:43]: Can you remove these lines of code and just print a warning to the console?

File: src/models/Train_Extra.java [hunk 5]
```diff
@@ -4,13 +4,13 @@
...
-    private final int trainId;
+    private final int manufactureYear;
     private final int numSeats;
     private final int numCars;
 
-    public Train_Extra(String model, int trainId, int numSeats, int numCars) {
+    public Train_Extra(String model, int manufactureYear, int numSeats, int numCars) {
         this.model = model;
-        this.trainId = trainId;
+        this.manufactureYear = manufactureYear;
```
Comment (anthonylopes54) [inline src/models/Train_Extra.java:13]: Nice fix

File: src/database/ModelHandler.java (hunk 2 above)
Comment (arcamelia) [reply src/database/ModelHandler.java:8]: you fool

File: src/database/StationHandler.java (hunk 3 above)
Comment (arcamelia) [reply src/database/StationHandler.java:50]: yea I noticed that when I was looking over your code - let's talk about it this afternoon

**Pull request closed without merging**